### PR TITLE
Books review

### PR DESCRIPTION
**Pull request by vamshi-kura**
Please could you review and let me know the changes if required

### CODE REVIEW

File: project-one/application.py [hunk 1]
```diff
@@ -93,10 +93,29 @@ def book_search():
...
+def book_details(isbn):
+    b = Books.query.filter_by(isbn = isbn).first()
+    r = Reviews.query.filter_by(book_isbn = isbn)
+    if request.method == 'POST':
+        re = request.form.get('post-review-data')
+        re1 = request.form.get('rating-value')
+        try:
+            if re != '' and  re1 != None :
+                rev = Reviews(book_isbn = b.isbn, user_name = session.get("USERNAME"), rating = re1, review = re)
+                db.session.add(rev)
```
Comment (ashishk144) [inline project-one/application.py:106]: where's the validation to see that the user has already entered a review or not?

Comment (ashishk144) [reply project-one/application.py:106]: how are you sure that the reviews are added into the review table and not any other table?

Comment (vamshi-kura) [reply project-one/application.py:106]: > where's the validation to see that the user has already entered a review or not?

The review table in models.py has a attribute unique for the user_name and book_isbn so when user again tries to review it results an exception so used try to except block to handle it and exception u can see the flash message.

**Comment by vamshi-kura** [comment]
> please post a screenshot of your rendered table.

Screenshot 2020-04-24 at 12 55 36 PM

File: project-one/application.py [hunk 2]
```diff
@@ -93,10 +93,29 @@ def book_search():
 
 
 # the book details page
-@app.route("/books/<isbn>")
-def book_details(isbn,title):
-    print(isbn, title)
-    # return render_template('book.html', )
+@app.route("/books/<isbn>" , methods = ['POST', 'GET'])
+def book_details(isbn):
+    b = Books.query.filter_by(isbn = isbn).first()
```
[the remark this answers is not in the document]
Comment (vamshi-kura) [reply project-one/application.py:98]: i'l change the calls and again push

**Comment by ashishk144** [comment]
> > please post a screenshot of your rendered table.
> 
> Screenshot 2020-04-24 at 12 55 36 PM

I meant HTML page and not adminer page.

File: project-one/application.py (hunk 1 above)
Comment (ashishk144) [reply project-one/application.py:106]: nice way to handle it

File: project-one/application.py [hunk 3]
```diff
@@ -75,6 +74,50 @@ def profile():
...
+        print(str(e))
+        flash('No data for the search', 'danger')
+        return redirect(url_for('profile'))
+
+
+# the book details page
+@app.route("/books/<isbn>" , methods = ['POST', 'GET'])
+def book_details(isbn):
+    # b = Books.query.filter_by(isbn = isbn).first()
+    # r = Reviews.query.filter_by(book_isbn = isbn)
```
Comment (ashishk144) [inline project-one/application.py:99]: remove commented code

**Pull request merged**